src/linter_test.cc

```diff
@@ -16,40 +16,100 @@
-    EXPECT_TRUE(checkLineLength(sql, 10, ' ').ok());
+    EXPECT_TRUE(CheckLineLength(sql).ok());
+    EXPECT_FALSE(CheckLineLength(sql, 10).ok());
+    EXPECT_TRUE(CheckLineLength(sql, 10, ' ').ok());
 }
 
-
 TEST(LinterTest, StatementValidityCheck) {
     const absl::string_view valid_sql = "SELECT 5+2";
 
     const absl::string_view invalid_sql = "SELECT 5+2 sss ddd";
 
-    EXPECT_TRUE(checkStatement(valid_sql).ok());
-    EXPECT_FALSE(checkStatement(invalid_sql).ok());
+    EXPECT_TRUE(CheckStatement(valid_sql).ok());
+    EXPECT_FALSE(CheckStatement(invalid_sql).ok());
 
-    EXPECT_TRUE(checkStatement(
+    EXPECT_TRUE(CheckStatement(
         "SELECT * FROM emp where b = a or c < d group by x").ok());
 
-    EXPECT_TRUE(checkStatement(
+    EXPECT_TRUE(CheckStatement(
         "SELECT e, sum(f) FROM emp where b = a or c < d group by x").ok());
 
-    EXPECT_FALSE(checkStatement(
+    EXPECT_FALSE(CheckStatement(
         "SELET A FROM B\nSELECT C FROM D").ok());
 }
 
-}  // namespace linter
-}  // namespace zetasql
+TEST(LinterTest, SemicolonCheck) {
+    const absl::string_view valid_sql =
+        "SELECT 3+5;\nSELECT 4+6;";
+    const absl::string_view invalid_sql =
+        "SELECT 3+5\nSELECT 4+6;";
+    const absl::string_view valid_sql2 =
+        "SELECT 3+5;  \nSELECT 4+6";
+    const absl::string_view invalid_sql2 =
+        "SELECT 3+5  ;  \nSELECT 4+6";
+
+    EXPECT_TRUE(CheckSemicolon(valid_sql).ok());
+    EXPECT_FALSE(CheckSemicolon(invalid_sql).ok());
+    EXPECT_TRUE(CheckSemicolon(valid_sql2).ok());
+    EXPECT_FALSE(CheckSemicolon(invalid_sql2).ok());
+}
+
+TEST(LinterTest, UppercaseKeywordCheck) {
+    const absl::string_view valid_sql =
+        "SELECT * FROM emp WHERE b = a OR c < d GROUP BY x";
+    const absl::string_view invalid_sql =
+        "SELECT * FROM emp where b = a or c < d GROUP by x";
+
+    EXPECT_TRUE(CheckUppercaseKeywords(valid_sql).ok());
+    EXPECT_FALSE(CheckUppercaseKeywords(invalid_sql).ok());
+}
+
+TEST(LinterTest, CommentTypeCheck) {
+    const absl::string_view valid_sql =
+        "//comment 1\nSELECT 3+5\n//comment 2";
+    const absl::string_view invalid_sql =
+        "//comment 1\nSELECT 3+5\n--comment 2";
+    const absl::string_view valid_sql2 =
+        "--comment 1\nSELECT 3+5\n--comment 2";
+
+    const absl::string_view multiline_comment_sql =
+        "/* here is // and -- */SELECT 1+2 -- comment 2";
+
+    const absl::string_view multiline_string_sql =
+        "SELECT \"\"\"multiline\nstring--\nliteral\nhaving//--//\"\"\"";
+
+    EXPECT_TRUE(CheckCommentType(valid_sql).ok());
+    EXPECT_FALSE(CheckCommentType(invalid_sql).ok());
+    EXPECT_TRUE(CheckCommentType(valid_sql2).ok());
+    EXPECT_TRUE(CheckCommentType(multiline_comment_sql).ok());
+    EXPECT_TRUE(CheckCommentType(multiline_string_sql).ok());
+}
+
+TEST(LinterTest, AliasKeywordCheck) {
+    const absl::string_view valid_sql =
+        "SELECT * FROM emp AS X";
+    const absl::string_view invalid_sql =
```

Comment:
add a test case "SELECT 1 one" (should be "SELECT 1 AS one")

Reply:
Added.